extensions/vscode-extensiontest/src/extension.ts

```diff
@@ -0,0 +1,58 @@
+import { ExtensionContext, ReferenceProvider, ReferenceContext, TextDocument, CancellationToken, ProviderResult, Location, Position, Uri, languages } from 'vscode';
+
+export function activate(context: ExtensionContext) {
+
+	const goReferenceProvider = new GoReferenceProvider();
+	languages.registerReferenceProvider("go", goReferenceProvider);
+	context.subscriptions.push(goReferenceProvider);
+}
+
+export class GoReferenceProvider implements ReferenceProvider {
+
+	provideReferences(document: TextDocument, position: Position, context: ReferenceContext, token: CancellationToken, progress: (locations: Location[]) => void): ProviderResult<Location[]> {
+		console.log("provide references");
+		return new Promise<Location[]>((resolve, reject) => {
```

Comment:
I don't think I can. The reduce promise is of type void and I need to return a promise of type Location[].

I could have the intermediate promises just always resolve with all locations, but that seems confusing.